api/maven-api-xml/src/main/java/org/apache/maven/api/xml/XmlNode.java

```diff
@@ -509,40 +506,45 @@ public int hashCode() {
             @Override
             public String toString() {
                 try {
-                    StringWriter writer = new StringWriter();
-                    XmlService.write(this, writer);
-                    return writer.toString();
-                } catch (IOException e) {
-                    return toStringObject();
+                    return XmlService.write(this, new StringWriter()).toString();
+                } catch (IOException ignored) {
+                    return addToStringField(
+                                    new StringJoiner(", ", "XmlNode[", "]"),
+                                    prefix,
+                                    namespaceUri,
+                                    name,
+                                    value,
+                                    attributes,
+                                    children,
+                                    inputLocation)
+                            .toString();
                 }
             }
 
-            private String toStringObject() {
```

Comment:
some consider interfaces to be public by default.

Reply:
Of course, it depends and might not apply to this situation. I would see anything in an interface as public, as that is the default scope. Interfaces are considered a public good, ready to be used.